app/scripts/modules/core/src/pipeline/config/triggers/pubsub/pubsub.trigger.ts

```diff
@@ -0,0 +1,44 @@
+import {IController, module} from 'angular';
+
+import {SETTINGS} from 'core/config/settings';
+import {PIPELINE_CONFIG_PROVIDER} from 'core/pipeline/config/pipelineConfigProvider';
+import {SERVICE_ACCOUNT_SERVICE, ServiceAccountService} from 'core/serviceAccount/serviceAccount.service';
+import {IPubsubTrigger} from '@spinnaker/core';
+
+class PubsubTriggerController implements IController {
+
+  public fiatEnabled: boolean = SETTINGS.feature.fiatEnabled;
+  public pubsubTypes = ['kafka', 'google'];
+  public serviceAccounts: string[] = [];
+
+  constructor(public trigger: IPubsubTrigger, private serviceAccountService: ServiceAccountService) {
+    'ngInject';
+
+    this.serviceAccountService.getServiceAccounts().then(accounts => {
```

Comment:
I don't see any field for selecting service accounts below, so I would omit this for now. I imagine that you'll have to do that eventually.